addons/project/models/project.py

```diff
@@ -324,6 +320,21 @@ def write(self, vals):
                 project.message_subscribe(project.partner_id.ids)
         return res
 
+    @api.multi
+    def unlink(self):
+        # Check project is empty
+        for project in self:
+            if project.tasks:
+                raise UserError(_('You cannot delete a project containing tasks. You can either archive it or first delete all of its tasks.'))
+        # Delete the empty related analytic account
+        analytic_accounts_to_delete = self.env['account.analytic.account']
+        for project in self:
+            if project.analytic_account_id and not project.analytic_account_id.line_ids:
+                analytic_accounts_to_delete |= project.analytic_account_id
+        result = super(Project, self).unlink()
+        analytic_accounts_to_delete.unlink()
+        return result
```

Comment:
can't we delete analytic_accounts before project ?

Reply:
Yes we could, but this is just code moves. So, this should be done during another task. Moreover, I think the condition if wrong. It will be fix after the task of "timesheet semantic"